pymeasure/instruments/ipgphotonics/yar.py

```diff
@@ -0,0 +1,182 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from enum import IntFlag
+
+from pymeasure.instruments import Instrument, validators
+from pyvisa.constants import Parity, StopBits
+
+
+def emission_validator(value, values):
+    if value in values:
+        return value
+    elif isinstance(value, str):
+        if value.capitalize() in values:
+            return value.capitalize()
```

Comment:
I think the first clause is redundant, as it's fully covered by the second?

Reply:
I streamlined the validator (first checks for boolean values, as that is the preferred way to interact).
With that change, I noticed a bug with the tests: it should be `upper`, not `capitalize`.

Now I test for a variety of input parameters.

Reply:
thanks!